### PR TITLE
feat: Add support for custom OpenAI API base URLs.

### LINKED ISSUE #105: feature: Custom OpenAI API address
I've built my own nginx to reverse proxy `api.openai.com` to bypass regional access restrictions.
Is it possible to add a configuration to sets the custom API server?

### PR DESCRIPTION
**Pull request by ferstar**
fix #105

### CODE REVIEW

**Review by j178 — APPROVED** [review]

**Comment by zurawiki** [comment]
Thank you for your contribution @ferstar. The PR looks good. Make sure to format your patch correctly using `just fix` (instructions in the Contributing.md file)

File: src/settings.rs [hunk 1]
```diff
@@ -281,6 +284,12 @@ impl Settings {
         settings = settings.add_source(app_env);
 
         // add custom override
+        if let Ok(openai_api_base) = std::env::var("OPENAI_API_BASE") {
```
Comment (zurawiki) [inline src/settings.rs:287]: I assume this ENVVAR is pretty standard across OpenAI tools. Looks good to me

**Review by zurawiki — APPROVED** [review]

**Comment by zurawiki** [comment]
Code looks good. Reformat your code and I can rebase it
```
just fix
```

File: src/settings.rs (hunk 1 above)
Comment (ferstar) [reply src/settings.rs:287]: ok, I'll do it

**Pull request merged**